tokio/src/io/driver/mod.rs

```diff
@@ -259,8 +259,7 @@ cfg_rt! {
         /// This function panics if there is no current reactor set and `rt` feature
         /// flag is not enabled.
         pub(super) fn current() -> Self {
-            crate::runtime::context::io_handle()
-                .expect("there is no reactor running, must be called from the context of Tokio runtime")
+            crate::runtime::context::io_handle().expect("io not enabled on Tokio context")
```

Comment:
This error should check whether there is a runtime context or not. If there is, it should fail with an error saying that there is a runtime, but IO is disabled on it. Otherwise it should fail with the same error as the other panics.

Reply:
@Darksonn I made `runtime::context::io_handle()` fail with the standard error if the Tokio context isn't instantiated, does this address your concern or should it be implemented differently?

If  `runtime::context::io_handle()` shouldn't panic in case of missing Tokio context, the return value has to differentiate between missing context and io being disabled.

Reply:
I can't tell there being any technical drawback to have `runtime::context::io_handle()` panic if the Tokio context is missing.

Reply:
It should panic in both cases, but the error messages should be different.

Reply:
@Darksonn I'm wondering if there might be a misunderstanding; the way I've implemented it, there are different error messages depending on whether the Tokio context is missing or io isn't enabled on it.

I implemented it so that `runtime::context::io_handle` will [panic](https://github.com/aknuds1/tokio/blob/chore/context-missing/tokio/src/runtime/context.rs#L18) if the context is missing. Otherwise, `io::driver::Handle::current` will [panic](https://github.com/aknuds1/tokio/blob/chore/context-missing/tokio/src/io/driver/mod.rs#L262) if the context hasn't io enabled.